proto/xmtpv4/message_api/message_api.proto

```diff
@@ -81,27 +81,22 @@ message MisbehaviorReport {
 }
 
 // Query for envelopes, shared by query and subscribe endpoints
+// Either topics or originator_node_ids may be set, but not both
 message EnvelopesQuery {
-  oneof filter {
-    // Client queries
-    bytes topic = 1;
-    // Node queries
-    uint32 originator_node_id = 2;
-  }
+  // Client queries
+  repeated bytes topics = 1;
```

Comment:
We don't want this to be a `oneof` anymore? What happens if I specify both topics and originator node ids?

Reply:
Still want this to be a `oneof`, with the server throwing an error if you specify both. Also left a comment above.

A `repeated` field cannot be placed inside a `oneof` field, so I decided to remove the `oneof filter` as the alternative is multiple layers of nesting - but could be convinced otherwise

Reply:
Ahh, missed the comment. I'm OK with this. The amount of nesting we have in our protos is pretty annoying to serialize/deserialize.